examples/static_react_task/run_task.py

```diff
@@ -4,27 +4,76 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+from mephisto.abstractions.blueprints.mixins.screen_task_required import (
+    ScreenTaskRequired,
+)
+from mephisto.data_model.unit import Unit
 from mephisto.operations.operator import Operator
 from mephisto.tools.scripts import task_script, build_custom_bundle
 from mephisto.abstractions.blueprints.abstract.static_task.static_blueprint import (
     SharedStaticTaskState,
 )
-
+from rich import print
 from omegaconf import DictConfig
 
 
-@task_script(default_config_file="example")
+def my_screening_unit_generator():
+    while True:
+        yield {"text": "SCREENING UNIT: Press the red button", "is_screen": True}
+
+
+def validate_screening_unit(unit: Unit):
+    agent = unit.get_assigned_agent()
+    if agent is not None:
+        data = agent.state.get_data()
+        print(data)
+        if (
+            data["outputs"] is not None
+            and "rating" in data["outputs"]
+            and data["outputs"]["rating"] == "bad"
+        ):
+            # User pressed the red button
+            return True
+    return False
+
+
+def onboarding_always_valid(onboarding_data):
+    return True
+
+
+@task_script(default_config_file="example.yaml")
 def main(operator: Operator, cfg: DictConfig) -> None:
-    def onboarding_always_valid(onboarding_data):
-        return True
-
-    shared_state = SharedStaticTaskState(
-        static_task_data=[
-            {"text": "This text is good text!"},
-            {"text": "This text is bad text!"},
-        ],
-        validate_onboarding=onboarding_always_valid,
-    )
+    is_using_screening_units = cfg.mephisto.blueprint["use_screening_task"]
+    shared_state = None
+    if not is_using_screening_units:
+        shared_state = SharedStaticTaskState(
+            static_task_data=[
+                {"text": "This text is good text!"},
+                {"text": "This text is bad text!"},
+            ],
+            validate_onboarding=onboarding_always_valid,
+        )
+
+    else:
+        """
+        When using screening units there has to be a
+        few more properties set on shared_state
+        """
+        shared_state = SharedStaticTaskState(
+            static_task_data=[
+                {"text": "This text is good text!"},
+                {"text": "This text is bad text!"},
+            ],
+            validate_onboarding=onboarding_always_valid,
+            on_unit_submitted=ScreenTaskRequired.create_validation_function(
+                cfg.mephisto,
+                validate_screening_unit,
+            ),
+            screening_data_factory=my_screening_unit_generator(),
+        )
+        shared_state.qualifications += ScreenTaskRequired.get_mixin_qualifications(
+            cfg.mephisto, shared_state
+        )
```

Comment:
If you are using the screening unit conf they are more properties that you need to add to your `SharedStaticTaskState`.

Reply:
You can add these dynamically to the `SharedStaticTaskState` after the case:
```
share_state = ...
if is_using_screening_units:
    shared_state.on_unit_submitted = ...
    ...

```
To reduce duplication a bit